plugins/fields.yaml

```diff
@@ -0,0 +1,78 @@
+apiVersion: krew.googlecontainertools.github.com/v1alpha2
+kind: Plugin
+metadata:
+  name: fields
+spec:
+  platforms:
+  - uri: https://github.com/rewanth1997/kubectl-fields/releases/download/v1.0.0/kubectl-fields-Darwin-x86_64.tar.gz
+    sha256: 2e163e28075a69d4d1dca2b338ec889d967b38fb3e0ed2de1c89993b99ae2889
+    bin: kubectl-fields
+    files:
+    - from: "*"
+      to: "."
+    selector:
+      matchLabels:
+        os: darwin
+        arch: amd64
+  - uri: https://github.com/rewanth1997/kubectl-fields/releases/download/v1.0.0/kubectl-fields-Linux-x86_64.tar.gz
+    sha256: 951739ca1c73dac0729ec46828e677effa80b607efbbf4e79829f7a243126af7
+    bin: kubectl-fields
+    files:
+    - from: "*"
+      to: "."
+    selector:
+      matchLabels:
+        os: linux
+        arch: amd64
+  - uri: https://github.com/rewanth1997/kubectl-fields/releases/download/v1.0.0/kubectl-fields-Windows-x86_64.tar.gz
+    sha256: 2785c6ccb4b1895c37423192f988dca4f3d4f8222cdce9fa0fdf60810e7c4eda
+    bin: kubectl-fields.exe
+    files:
+    - from: "*"
+      to: "."
+    selector:
+      matchLabels:
+        os: windows
+        arch: amd64
+  version: v1.0.0
+  homepage: https://github.com/rewanth1997/kubectl-fields
+  caveats: |
+    Usage:
+      $ kubectl fields [options] <RESOURCE> <PATTERN>
+
+    Options:
+      -h, --help          help for kubectl-fields
+      -i, --ignore-case   Ignore case distinction
```

Comment:
What does this apply to? The search pattern or the resource? If the pattern, then I suggest you change the example below from `kubectl fields -i svc ip` to `kubectl fields svc -i ip` (it confused me when I first saw the example).

Reply:
Ah, and ignore-case should probably be true by default, with the option to turn it off. I think it would make the UX better (after all, does a user searching for a field know how the field is capitalized?)

Reply:
`-i` is a non-positional parameter. It can be used anywhere. I think just changing the example in the manifest will resolve the issue.

`kubectl fields -i svc ip` == `kubectl fields svc -i ip` == `kubectl fields svc ip -i`

Coming to the second question, I observed if the default value of `ignore-case` is true it will return lots of results that might not be required. Also, I was following `grep` convention just to make it easy for the users. Please let us know what you think.

```
rewanth@ubuntu:/$ kubectl fields daemonset -i ver
apiVersion
metadata.initializers.result.apiVersion
metadata.initializers.result.metadata.resourceVersion
metadata.managedFields.apiVersion
metadata.ownerReferences.apiVersion
metadata.resourceVersion
spec.template.metadata.initializers.result.apiVersion
spec.template.metadata.initializers.result.metadata.resourceVersion
spec.template.metadata.managedFields.apiVersion
spec.template.metadata.ownerReferences.apiVersion
spec.template.metadata.resourceVersion
spec.template.spec.containers.env.valueFrom.fieldRef.apiVersion
spec.template.spec.dnsConfig.nameservers
spec.template.spec.initContainers.env.valueFrom.fieldRef.apiVersion
spec.template.spec.volumes.csi.driver
spec.template.spec.volumes.downwardAPI.items.fieldRef.apiVersion
spec.template.spec.volumes.flexVolume.driver
spec.template.spec.volumes.iscsi.chapAuthDiscovery
spec.template.spec.volumes.nfs.server
spec.template.spec.volumes.projected.sources.downwardAPI.items.fieldRef.apiVersion
```
```
rewanth@ubuntu:/$ kubectl fields daemonset ver
spec.template.spec.dnsConfig.nameservers
spec.template.spec.volumes.csi.driver
spec.template.spec.volumes.flexVolume.driver
spec.template.spec.volumes.iscsi.chapAuthDiscovery
spec.template.spec.volumes.nfs.server
```